src/NuGet.Core/NuGet.Protocol/Utility/CachingUtility.cs

```diff
@@ -12,6 +12,8 @@ namespace NuGet.Protocol
     public class CachingUtility
     {
         public const int BufferSize = 8192;
+        // To maintain SHA-1 backwards compatibility with respect to the length of the hex-encoded hash, the hash will be truncated to a length of 20 bytes.
```

Comment:
Why do we need to maintain the 20 byte length? THe actual hashes are going to be different anyway, so I don't understand the benefit.

Reply:
As the SHA2 is 256 bits and SHA1 is 160 bits, the hash string will be 24 characters longer. The hash string will be part of the lock file path. So it's safer to truncate it as not to exceed the max length of path (260 charactors)

Reply:
The location on my Windows 10 machine for the http cache is `C:\Users\<user_id>\AppData\Local\NuGet\v3-cache\`. This is 8 characters shorter than the path used by `ConcurrencyUtilities`, and therefore allows the `<user_id>` to be 8 characters longer than the numbers I wrote in that comment.

Reply:
Keep in mind that the location of the HTTP cache is [configurable](https://docs.microsoft.com/en-us/nuget/consume-packages/managing-the-global-packages-and-cache-folders).  We should avoid breakages from unnecessarily increasing path length requirements.